packages/effects-core/src/math/shape/split-union.ts

```
@@ -0,0 +1,153 @@
+  return output.filter(c => c.length >= 6);
+}
+
+// 拆分：自交区域分开
+export const split = (c: number[]) => resolveContour(c, 'ODD');
+
+// 并集：自交区域合并为整体外轮廓
+// 如果有分离的部分，会输出多个
+export const union = (c: number[]) => resolveContour(c, 'NONZERO');
+
+// 最大几何区域：自交拆分后最大的一块
+export function splitMax(contour: number[]): number[]
```

ESLint

[error] Missing space before function parentheses.

(`@typescript-eslint/space-before-function-paren`)

---

[error] Opening curly brace does not appear on the same line as controlling statement.

(brace-style)